### ISSUE TITLE
Issue with using callback functions

### ISSUE THREAD

**Issue by galliumsoft** [styled-components/styled-components#1179]
Not sure if it is a new feature request or an issue or a plain simple wrong way of doing it... but thought will check neverthless.. Am facing an issue with trying to send a callback function into a styled-component.. the callback then updates a state in the parent function.. (it crashes, obviously!)... So am wondering if there is a way out here.. The use-case i want to achieve is as below.. (pasting only the relevant sections of code)

in parent.. 
--------------------
handledrawerhover = ()=> {this.setState({hovered: !this.state.hovered})

render() {
    return (
      <Styled.Drawer
        className = "MainDrawer"
        handleDrawerHover={this.handleDrawerHover} >{children}</Styled.Drawer>
-----------------------
and in the styled component function.. 
---------------
import Drawer from 'material-ui/Drawer'

const Styled = {
  Drawer: styled(Drawer) `
      &:hover {
        ${props=>props.handleDrawerHoverEvent};
      }
  `,
}
----------------------
      Basically the output i want to achieve is the state to be set on hover on the Drawer element. But looks like there is no way i can execute a callback function.. or is there?

**Comment by diondiondion** [comment]
That's not something CSS can do. You need to handle this in JS, for example using the `onMouseEnter` event:
```jsx
<Styled.Drawer
    onMouseEnter={this.handleDrawerHover}
>
    {children}
</Styled.Drawer>
```

(It only fires when the cursor enters the element. Depending on what `this.handleDrawerHover` does, you may want to use the `onMouseLeave` event to handle that case, too.)

By the way, put code into backticks to make it more readable:
````
```
This will be displayed as code.
```
````

**Comment by mxstbr** [comment]
Thanks for answering @diondiondion!
